src/bin/agrind.rs

```diff
@@ -42,15 +44,35 @@ pub enum InvalidArgs {
     MissingQuery,
 }
 
+/// An ErrorReporter that writes errors related to the query string to the terminal
+struct TermErrorReporter {
+    formatter: annotate_snippets::formatter::DisplayListFormatter,
+}
+
+impl ErrorReporter for TermErrorReporter {
+    fn handle_error(&self, snippet: Snippet) {
+        let dl = annotate_snippets::display_list::DisplayList::from(snippet);
+
+        eprintln!("{}", self.formatter.format(&dl));
+    }
+}
+
 fn main() -> CliResult {
     setup_panic!();
     let args = Cli::from_args();
     if args.update {
         return update();
     }
-    let query = &args.query.ok_or(InvalidArgs::MissingQuery)?;
+    let query = QueryContainer::new(
+        args.query.ok_or(InvalidArgs::MissingQuery)?,
+        Box::new(TermErrorReporter {
+            formatter: annotate_snippets::formatter::DisplayListFormatter::new(
+                env::var("NO_COLOR").is_err(),
```

Comment:
Ah, yes, didn't think to check if it's a TTY here.  I ended up using the atty crate since the terminal_size crate only looks at stdout and not stderr.  I was thinking we would still want colored errors if the user redirected the output to a file.

The NO_COLOR env var seems to be an existing convention, see https://no-color.org/ .  So, I've left that in as well.